src/lib/crawler/crawler.ts

```diff
@@ -1,19 +1,47 @@
 import axios from "axios";
+import * as cheerio from "cheerio";
 
+// defining Crawler interface
 export interface Crawler {
   getMatchIds(params: { character: string; realm: string }): Promise<string[]>;
-  getMatchDetails(params: { matchIds: string[] }): Promise<any[]>;
+  getMatchDetails(params: { matchIds: string[] }): Promise<any[]>; //
 }
-
+// implementing the Crawler interface in the WarmaneCrawler class
 export class WarmaneCrawler implements Crawler {
+  // defining 'getMatchIds' method with optional paramater for testing (html)
   async getMatchIds(params: {
     character: string;
     realm: string;
+    // added optional paramater (HTML passed as string) for testing
+    html?: string;
```

Comment:
So generally we want to avoid having optional parameters in function signatures just for testing purposes. Your code does actually work though. There was a defect in my unit test which probably caused a lot of your confusion and changes.

When you use the `html` parameter when calling `getMatchIds()`, it won't use the axios call at all, therefore it is not tested at all. You can see this in code coverage tools.

![image](https://user-images.githubusercontent.com/11894935/228065711-84484285-86fa-4c9d-9198-1e149c89a5c0.png)

It's kind of hard to make sense of that at first glance but the line is yellow is indicating a partial test coverage i.e. one side of the nullish coalessence operator is tested but the other is not.

The red line indicates that there is no test coverage. In my opinion, I think it should've highlighted line 21 through 25 as red because it's all one expression, but JavaScript and it's tooling is weird sometimes.

The defect in my unit test was that I was returning a string in my mock axios implementation where I should've been returning an axios response object with a property data and string value.

It was:
```TypeScript
    mockedAxios.get.mockImplementation(async () => testHtml);
```

when it should've been

```TypeScript
    mockedAxios.get.mockImplementation(async () => ({
      data: testHtml,
    }));
```

I request that we remove the optional `html` parameter and correct the mock implementation.

Finally, we have happened upon some inconvenient timing. Apparently the arena season ended on warmane and the data coming back no longer matches what is in my test file. This was inevitable, but it is kind of inconvenient for short term testing purposes. Where I had 453 games played, it now says I have 1.